ExampleApp/ExampleApp/Screens/Characteristics/ViewController/CharacterisiticsViewController.swift

```diff
@@ -0,0 +1,175 @@
+import RxBluetoothKit
+import RxSwift
+import RxCocoa
+import UIKit
+
+final class CharacteristicsViewController: UIViewController, CustomView {
+
+    typealias ViewClass = BaseView
+
+    typealias CharacteristicsDataSource = TableViewDataSource<[Characteristic], CharacteristicsViewModelItem>
+
+    private let viewModel: CharacteristicsViewModelType
+
+    private let dataSource: CharacteristicsDataSource
+
+    private let disposeBag: DisposeBag = DisposeBag()
+
+    init(with dataSource: CharacteristicsDataSource, viewModel: CharacteristicsViewModelType) {
+        self.dataSource = dataSource
+        self.viewModel = viewModel
+        super.init(nibName: nil, bundle: nil)
+        view.backgroundColor = .white
+    }
+
+    @available(*, unavailable)
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func loadView() {
+        self.view = ViewClass()
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setupTableView()
+        setDataSourceRefreshBlock()
+        bindViewModel()
+    }
+
+    private func setupTableView() {
+        customView.setTableView(dataSource: dataSource, delegate: self)
+        registerCells()
+    }
+
+    private func setDataSourceRefreshBlock() {
+        self.dataSource.refreshDataBlock = { [weak self] in
+            self?.customView.refreshTableView()
+        }
+    }
+
+    private func bindViewModel() {
+        subscribeViewModelOutputs()
+        dataSource.bindItemsObserver(to: viewModel.characteristicsOutput)
+    }
+
+    private func subscribeViewModelOutputs() {
+        subscribeDataUpdateOutput()
+        subscribeAlertTriggerOutput()
+    }
+
+    private func subscribeDataUpdateOutput() {
+        viewModel.dataUpdateOutput.subscribe(onNext: { [unowned self]  _ in
+            self.customView.refreshTableView()
+        }).disposed(by: disposeBag)
+    }
+
+    private func subscribeAlertTriggerOutput() {
+        viewModel.alertTriggerOutput.subscribe(onNext: { [unowned self] message in
+            self.showAlert(title: Constant.Strings.titleSuccess, message: message)
+        }).disposed(by: disposeBag)
+    }
+
+    private func registerCells() {
+        customView.register(cellType: CharacterisiticsTableViewCell.self,
+                forCellReuseIdentifier: String(describing: CharacterisiticsTableViewCell.self))
+    }
+
+    private func addWriteActions(to actionSheet: UIAlertController) {
+        let writeValueNotificationAction = UIAlertAction(title: Constant.Strings.titleWrite,
+                style: .default) { _ in
+            self.showWriteFieldForCharacteristic()
+        }
+        actionSheet.addAction(writeValueNotificationAction)
+    }
+
+    private func addReadActions(to actionSheet: UIAlertController) {
+        let readValueNotificationAction = UIAlertAction(title: Constant.Strings.titleRead,
+                style: .default) { [weak self] _ in
```

Comment:
no need for `[weak self]` since `readValueNotificationAction` is not captured as property of self